### ISSUE TITLE
Null pointer exception when trying to save session due to genome

### ISSUE THREAD

**Issue by julianstanley** [igvteam/igv#1330]
On version 2.16.1 on a M1 Mac

A previously-loaded genome was causing a null pointer exception when I tried to save a session.

I noticed that, when I navigated to a different genome, I could save just fine. But, with a certain genome, I was getting an error.

So, I re-loaded that genome file. Then, I was able to save. Now I have two copies of that genome listed (NC_005350_2.fna):

<img width="418" alt="image" src="https://github.com/igvteam/igv/assets/22749289/35833a94-6716-46de-a48a-c42bf7c33906">

My first thought was that maybe the path I originally specified for that fna file had changed, and that was causing the null pointer. 

However, here is a portion of my `~/igv/user-defined-genomes.txt` file:

```
[...]	NC_000913
NC_004350_2.fna	../../../../../../../Volumes/li/labshare/genome_annotations/Smutans/NCBI/NC_004350_2.fna	../../../../../../../Volumes/li/labshare/genome_annotations/Smutans/NCBI/NC_004350_2.fna
```

So these look the same, right?

Anyways, everything is all good for me after re-loading the genome (I just have to use the top one, not the bottom), but this might be a bug, or at least an opportunity for a better error. And maybe related to #167 

-----------

Original sequence of events:

1. I load a previously-saved xml snapshot
2. I load some new tracks
3. I click File-->Save Session
4. After saving, I get "Error creating session" popup, then an "There was an error writing to <filename> (java.lang.NullPointerException).

-----------

Log:

```
SEVERE [May 15,2023 17:42] [SaveSessionMenuAction] java.lang.RuntimeException: java.lang.NullPointerException
	at org.igv/org.broad.igv.session.SessionWriter.createXmlFromSession(SessionWriter.java:183)
	at org.igv/org.broad.igv.session.SessionWriter.saveSession(SessionWriter.java:91)
	at org.igv/org.broad.igv.ui.IGV.saveSession(IGV.java:1059)
	at org.igv/org.broad.igv.ui.action.SaveSessionMenuAction.actionPerformed(SaveSessionMenuAction.java:105)
	at java.desktop/javax.swing.AbstractButton.fireActionPerformed(Unknown Source)
	at java.desktop/javax.swing.AbstractButton$Handler.actionPerformed(Unknown Source)
	at java.desktop/javax.swing.DefaultButtonModel.fireActionPerformed(Unknown Source)
	at java.desktop/javax.swing.DefaultButtonModel.setPressed(Unknown Source)
	at java.desktop/javax.swing.AbstractButton.doClick(Unknown Source)
	at java.desktop/com.apple.laf.ScreenMenuItem.actionPerformed(Unknown Source)
	at java.desktop/java.awt.MenuItem.processActionEvent(Unknown Source)
	at java.desktop/java.awt.MenuItem.processEvent(Unknown Source)
	at java.desktop/java.awt.MenuComponent.dispatchEventImpl(Unknown Source)
	at java.desktop/java.awt.MenuComponent.dispatchEvent(Unknown Source)
	at java.desktop/java.awt.EventQueue.dispatchEventImpl(Unknown Source)
	at java.desktop/java.awt.EventQueue$4.run(Unknown Source)
	at java.desktop/java.awt.EventQueue$4.run(Unknown Source)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(Unknown Source)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(Unknown Source)
	at java.desktop/java.awt.EventQueue$5.run(Unknown Source)
	at java.desktop/java.awt.EventQueue$5.run(Unknown Source)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(Unknown Source)
	at java.desktop/java.awt.EventQueue.dispatchEvent(Unknown Source)
	at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(Unknown Source)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(Unknown Source)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(Unknown Source)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(Unknown Source)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(Unknown Source)
	at java.desktop/java.awt.EventDispatchThread.run(Unknown Source)
Caused by: java.lang.NullPointerException
	at org.igv/org.broad.igv.util.FileUtils.getRelativePath(FileUtils.java:135)
	at org.igv/org.broad.igv.util.FileUtils.getRelativePath(FileUtils.java:196)
	at org.igv/org.broad.igv.session.SessionWriter.createXmlFromSession(SessionWriter.java:127)
	... 30 more
```

**Comment by alefrol638** [comment]
For me the problem was solved, when I replaced all relative paths in the original xml session file with absolute paths. here the first lines of the xml session file, which caused the error : 
```
<?xml version="1.0" encoding="UTF-8" standalone="no"?>
<Session genome="../../genome/genome.fa" hasGeneTrack="true" hasSequenceTrack="true" locus="All" version="8">
	<Resources>
		<Resource path="../../bwa/mergedReplicate/bigwig/Teff_KI.mRp.clN.bigWig"/>
		<Resource path="../../bwa/mergedReplicate/macs/broadPeak/Teff_KI.mRp.clN_peaks.broadPeak"/>
		<Resource path="../../bwa/mergedLibrary/bigwig/Teff_KI_R1.mLb.clN.bigWig"/>
		<Resource path="../../bwa/mergedLibrary/macs/broadPeak/Teff_KI_R1.mLb.clN_peaks.broadPeak"/>
		<Resource path="../../bwa/mergedLibrary/bigwig/Teff_KI_R2.mLb.clN.bigWig"/>
		<Resource path="../../bwa/mergedLibrary/macs/broadPeak/Teff_KI_R2.mLb.clN_peaks.broadPeak"/>
		<Resource path="../../bwa/mergedLibrary/bigwig/Teff_KI_R3.mLb.clN.bigWig"/>
		<Resource path="../../bwa/mergedLibrary/macs/broadPeak/Teff_KI_R3.mLb.clN_peaks.broadPeak"/>
		<Resource path="../../bwa/mergedLibrary/bigwig/Teff_KI_R4.mLb.clN.bigWig"/>
		<Resource path="../../bwa/mergedLibrary/macs/broadPeak/Teff_KI_R4.mLb.clN_peaks.broadPeak"/>
		<Resource path="../../bwa/mergedReplicate/bigwig/Teff_WT.mRp.clN.bigWig"/>
		<Resource path="../../bwa/mergedReplicate/macs/broadPeak/Teff_WT.mRp.clN_peaks.broadPeak"/>
		<Resource path="../../bwa/mergedLibrary/bigwig/Teff_WT_R1.mLb.clN.bigWig"/>
		<Resource path="../../bwa/mergedLibrary/macs/broadPeak/Teff_WT_R1.mLb.clN_peaks.broadPeak"/>
		<Resource path="../../bwa/mergedLibrary/bigwig/T
```

When I replaced "../.." with the absolute paths(E.g on MacOS: "/Users/user/path_to_folder"), using find & replace, I was able to save and open the changed xml session file:

```
<?xml version="1.0" encoding="UTF-8" standalone="no"?>
<Session genome="/Users/user/path-to-folder/genome/genome.fa" hasGeneTrack="true" hasSequenceTrack="true" locus="All" version="8">
	<Resources>
		<Resource path="/Users/user/path-to-folder/bwa/mergedReplicate/bigwig/Teff_KI.mRp.clN.bigWig"/>
		<Resource path="/Users/user/path-to-folder/bwa/mergedReplicate/macs/broadPeak/Teff_KI.mRp.clN_peaks.broadPeak"/>
		<Resource path="/Users/user/path-to-folder/bwa/mergedLibrary/bigwig/Teff_KI_R1.mLb.clN.bigWig"/>
		<Resource path="/Users/user/path-to-folder/bwa/mergedLibrary/macs/broadPeak/Teff_KI_R1.mLb.clN_peaks.broadPeak"/>
		<Resource path="/Users/user/path-to-folder/bwa/mergedLibrary/bigwig/Teff_KI_R2.mLb.clN.bigWig"/>
		<Resource path="/Users/user/path-to-folder/bwa/mergedLibrary/macs/broadPeak/Teff_KI_R2.mLb.clN_peaks.broadPeak"/>
		<Resource path="/Users/user/path-to-folder/bwa/mergedLibrary/bigwig/Teff_KI_R3.mLb.clN.bigWig"/>
		<Resource path="/Users/user/path-to-folder/bwa/mergedLibrary/macs/broadPeak/Teff_KI_R3.mLb.clN_peaks.broadPeak"/>
		<Resource path="/Users/user/path-to-folder/bwa/mergedLibrary/bigwig/Teff_KI_R4.mLb.clN.bigWig"/>
		<Resource path="/Users/user/path-to-folder/bwa/mergedLibrary/macs/broadPeak/Teff_KI_R4.mLb.clN_peaks.broadPeak"/>
		<Resource path="/Users/user/path-to-folder/bwa/mergedReplicate/bigwig/Teff_WT.mRp.clN.bigWig"/>
		<Resource path="/Users/user/path-to-folder/bwa/mergedReplicate/macs/broadPeak/Teff_WT.mRp.clN_peaks.broadPeak"/>
		<Resource path="/Users/user/path-to-folder/bwa/mergedLibrary/bigwig/Teff_WT_R1.mLb.clN.bigWig"/>
		<Resource path="/Users/user/path-to-folder/bwa/mergedLibrary/macs/broadPeak/Teff_WT_R1.mLb.clN_peaks.broadPeak"/>
		<Resource path="/Users/user/path-to-folder/bwa/mergedLibrary/bigwig/Teff_WT_R2.mLb.clN.bigWig"/>
		<Resource path="/Users/user/path-to-folder/bwa/mergedLibrary/macs/broadPeak/Teff_WT_R2.mLb.clN_peaks.broadPeak"/>
		<Resource path="/Users/us
```
Hope, that this information helps for developing the next version. Thanks in advance!

**Comment by jrobinso** [comment]
OK thanks for the reports I will look into this.   I have been considering changing the default to "absolute" paths for sometime,  fewer issues such as this.  Relative paths are useful if you want to save sessions in folders with the data files and share the entire folder.   In some cases the folder can appear on a different path for different users,  relative paths fixes that problem but creates others, apparently.

**Comment by jrobinso** [comment]
Fixed with commit ee5ea52d.  This will be released by the end of the month in 2.17.0
